File: CodingInterviewsPackSolutionsCPP/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>sumMap;
        int sz =nums.size();
        for(int index = 0; index < sz; index++){
            int remainder = target-nums[index];
            if(sumMap.count(remainder))
                return{sumMap[remainder], index};
            
            sumMap[nums[index]] = index;
        }
        return {-1,-1};
    }
};

File: CodingInterviewsPythonSolutions/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sumMap = {}
        for index, num in enumerate(nums):
            remainder = target - num
            if remainder in sumMap:
                return [sumMap[remainder], index]
            
            sumMap[num] = index
        return [-1, -1]
